src/mappers.ts

```diff
@@ -1,515 +1,585 @@
+    if (criteria.indexOf(",") != -1) {
+      data.action = tempStr.trim();
+    } else {
+      data.condition = tempStr
+        .substring(0, util.indexOfFirstDigit(tempStr))
+        .trim();
+      data.value = tempStr.substr(util.indexOfFirstDigit(tempStr)).trim();
     }
+    ValidateAndAddCriteria(data);
+  });
+}
+function ValidateAndAddCriteria(data: any) {
+  if (data.action == "") data.action = "continue";
+  data.value = util.removeUnits(data.value);
+  if (!util.validCriteria(data)) throw new Error("Invalid Failure Criteria");
+  var key: string =
+    data.clientMetric +
+    " " +
+    data.aggregate +
+    " " +
+    data.condition +
+    " " +
+    data.action;
+  if (data.requestName != "") {
+    key = key + " " + data.requestName;
+  }
+  var val: number = parseInt(data.value);
+  var currVal = val;
+  if (failureCriteriaValue.hasOwnProperty(key))
+    currVal = failureCriteriaValue[key];
+  if (data.condition == ">") {
+    failureCriteriaValue[key] = val < currVal ? val : currVal;
+  } else {
+    failureCriteriaValue[key] = val > currVal ? val : currVal;
+  }
 }
 function getFailureCriteria(existingCriteriaIds: string[]) {
-    var numberOfExistingCriteria = existingCriteriaIds.length;
-    var index = 0;
-    for(var key in failureCriteriaValue) {
-        var splitted = key.split(" "); 
-        var criteriaId = index < numberOfExistingCriteria ? existingCriteriaIds[index++] : util.getUniqueId();
-        failCriteria[criteriaId] = {     
-            clientMetric: splitted[0],
-            aggregate: splitted[1],
-            condition: splitted[2],
-            action: splitted[3],
-            value: failureCriteriaValue[key],
-            requestName: splitted.length > 4 ? splitted.slice(4).join(' ') : null
-        };
-    }
-    for(; index < numberOfExistingCriteria; index++){
-        failCriteria[existingCriteriaIds[index]] = null;
-    }
-}
\ No newline at end of file
+  var numberOfExistingCriteria = existingCriteriaIds.length;
+  var index = 0;
+  for (var key in failureCriteriaValue) {
+    var splitted = key.split(" ");
+    var criteriaId =
+      index < numberOfExistingCriteria
+        ? existingCriteriaIds[index++]
+        : util.getUniqueId();
+    failCriteria[criteriaId] = {
+      clientMetric: splitted[0],
+      aggregate: splitted[1],
+      condition: splitted[2],
+      action: splitted[3],
+      value: failureCriteriaValue[key],
+      requestName: splitted.length > 4 ? splitted.slice(4).join(" ") : null,
+    };
+  }
+  for (; index < numberOfExistingCriteria; index++) {
+    failCriteria[existingCriteriaIds[index]] = null;
+  }
+}
+function getAutoStopCriteria() {
+  let data: autoStopCriteriaObj = {
+    isAutoAbortEnabled: true,
+    errorRate: 90,
+    errorRateTimeWindow: 60,
+  };
+
+  if (typeof autoStop == "string") {
+    data = {
+      isAutoAbortEnabled: false,
+      errorRate: 0,
+      errorRateTimeWindow: 0,
+    };
+  }
```

Comment:
nit: Should we add validation for the autoStop string value to be "disable" ? And fail with error that auto stop is not valid if it is not correct?

Reply:
updated